docs/site/Services.md

```diff
@@ -0,0 +1,112 @@
+---
+lang: en
+title: 'Services'
+keywords: LoopBack 4.0, LoopBack 4
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Services.html
+---
+
+## Overview
+
+We use the `service` concept in LoopBack 4 to refer to an object with methods to
+perform local or remote operations. Such objects are resolved from
+[bindings](Binding.md) within a [LoopBack 4 context](Context.md).
+
+We support three types of services:
+
+- proxy: A service proxy for remote REST/SOAP/gRPC APIs
+- class: A TypeScript class that can be bound to the application context
+- provider: A TypeScript class that implements `Provider` interface and can be
+  bound to the application context
+
+For remote services (type=proxy), there should be at least one valid _(REST,
+SOAP, or gRPC)_ data source created already in the `src/datasources` directory.
+
+## Generate services using CLI
+
+The `lb4 service` command can generate code for local and remote services.
+
+See [Service Generator](Service-generator.md) for more details.
+
+## Register services with the application
+
+Services added to `src/services` are automatically discovered and registered
+when the application is booted.
+
+To register a service by code, use
+[application.service()](https://loopback.io/doc/en/lb4/apidocs.core.application.service.html).
+For example,
+
+- Bind a service class or provider
+
+  ```ts
+  // For a class, the interface is default to the class itself
+  const binding = app.service(MyService);
+  ```
+
+  ```ts
+  // A custom name can be specified when the service is registered
+  const binding = app.service(MyService, 'my-service');
+  ```
+
+- Bind a service class or provider with an interface identified by a string
+
+  ```ts
+  // Register a service and mark the interface as `MyService`
+  const binding = app.service(MyService, {interface: 'MyService'});
+  ```
+
+  ```ts
+  // Register a service provider and mark the interface as `MyService`
+  const binding = app.service(MyServiceProvider, {interface: 'MyService'});
```

Comment:
Can you please explain the rules we are using to decide whether the supplied constructor is a service class or a provider class?